party_menu/party_menu.lua

```diff
@@ -0,0 +1,521 @@
+
+    return name_job_hp_format .. mp_format .. tp_format .. debuffs_format
+end
+
+local function check_target_for_row_highlighting()
+    local entity = target.t or target.st
+    for i = 1, 6 do
+        if entity and party[i] then
+            if entity.name == party[i].name then
+                party_window_states[i].color = ui.color.rgb(30, 30, 30)
+            else
+                party_window_states[i].color = ui.color.black
+            end
+        else
+            party_window_states[i].color = ui.color.black
+        end
+    end
+end
+
+-- Player zone out packet
+packets.incoming[0x00B]:register(function()
+    zone_out_time = os.clock()
+    zoning = true
+end)
+
+-- 0x0DF appears to change when player hp/mp/tp changes
+packets.incoming[0x0DF]:register(function()
+    update_party_name_hp_mp_tp_strings('both')
+    update_party_debuff_strings('both')
+end)
+
+-- This seems to change when buffs on self change, also when party members zone
+packets.incoming[0x037]:register(function()
+    update_party_name_hp_mp_tp_strings('both')
+    update_party_debuff_strings('both')
+end)
+
+-- Party member updates, joining/leaving party etc
+packets.incoming[0x0DD]:register(function()
+    update_party_name_hp_mp_tp_strings('both')
+    update_party_debuff_strings('both')
+end)
+
+-- Party buffs update packet
+packets.incoming[0x076]:register(function()
+    update_party_debuff_strings('party_only')
+end)
+
+ui.display(function()
+    if zoning and os.clock() > zone_out_time + zone_in_delay then
+        zoning = false
+    end
+    if not global_window_closed and not zoning then
+        update_ui_dimensions()
+        check_target_for_row_highlighting()
+        global_window_state, global_window_closed = ui.window('global_window', global_window_state, function()
+            for i = 1, 6 do
+                if party[i] then
+                    party_window_states[i] = ui.window('party_member_row', party_window_states[i], function()
+                        ui.location(left_inner_padding, upper_inner_padding)
+                        ui.text(party_display_strings[i].name_hp_mp_tp .. party_display_strings[i].debuffs)
+                    end)
+                else
+                    break
+                end
+            end
+        end)
+    end
+end)
+
+local pm = command.new('pm')
+
+local function help()
+    command.input('/echo Party Menu commands')
+    command.input('/echo /pm h | help')
+    command.input('/echo /pm t | tog | toggle')
+    command.input('/echo /pm name | names')
+    command.input('/echo /pm hp')
+    command.input('/echo /pm mp')
+    command.input('/echo /pm tp')
+    command.input('/echo /pm db | debuff | debuffs')
+    command.input('/echo /pm color')
+    command.input('/echo /pm size [new_size]')
+    command.input('/echo /pm pos | position [x] [y]')
+    command.input('/echo /pm wid | width [new_width]')
+    command.input('/echo /pm s | save')
+end
+
+pm:register('h', help)
+pm:register('help', help)
```

Comment:
This will also need to go, but may leave it in for now. Help text is automatically generated on `/pm` (or if you input any other incorrect command). `/pm help` currently doesn't work but I intend to add it shortly. I will also make it output to the chat log (currently it only prints to the console). You can do this though:
```lua
local help = function()
    print(pm:syntax())
end
```